conans/client/downloaders/caching_file_downloader.py

```diff
@@ -47,44 +47,48 @@ def _lock(self, lock_id):
-                self._file_downloader.download(download_url + h + ".json",
-                                               cached_path + ".json", **kwargs)
-                found_in_backup = True
-                conanfile.output.info(f"Sources from {url} found in remote backup {download_url}")
+                if download_url == origin_placeholder:
+                    self._file_downloader.download(url, cached_path, sha256=sha256, **kwargs)
+                else:
+                    self._file_downloader.download(download_url + h, cached_path, sha256=sha256,
+                                                   **kwargs)
+                    self._file_downloader.download(download_url + h + ".json", cached_path + ".json",
+                                                   **kwargs)
+                found = True
+                location = "origin" if download_url == origin_placeholder else "remote backup"
+                conanfile.output.info(f"Sources from {url} found in {location} {download_url}")
                 break
             except NotFoundException:
-                # TODO: What happens if sha256 missmatch?
-                pass
+                if download_url == origin_placeholder and download_urls.index(origin_placeholder) == 0:
+                    conanfile.output.warning(
+                        f"File could not be fetch from origin '{url}', trying with sources backups next")
             except (AuthenticationException, ForbiddenException) as e:
-                raise ConanException(f"The source backup server '{download_url}' needs authentication"
-                                     f"/permissions, please provide 'source_credentials.json': {e}")
-        if not found_in_backup:
-            policy = global_conf.get("core.backup_sources:cache_miss_policy", check_type=str,
-                                     default="ignore")
-            # TODO: Think about what is the best default
-            message = f"Sources from {url} not found in remote backup sources server(s)"
-            if policy == "error":
-                raise ConanException(message)
-            elif policy == "warn":
-                conanfile.output.warning(message)
-            elif policy == "ignore":
-                # Do nothing, we're fine with looking externally for the files
-                pass
-            else:
-                raise ConanException("Backup sources cache missed but "
-                                     "'core.backup_sources:cache_miss_policy' "
-                                     f"has an unknown value of '{policy}'")
-        return found_in_backup
+                raise ConanException(
+                    f"The source backup server '{download_url}' needs authentication"
+                    f"/permissions, please provide 'source_credentials.json': {e}")
+            except ConanException as e:
+                # TODO: Rethink how we detect a broken origin
```

Comment:
This is NOT how we should do it, but pushed with this to gather feedback on how to properly handle this. A 500 error is raised as a ConanException, but we don't want to catch _every_ ConanException as a pass if trying with internet 🤔 
Maybe a different Exception type such as NotFound and Authentication do? I'd have done like so from the begining, but I have my doubts that this could be a breaking behaviour in the tools.download